### DISCUSSION TITLE
Using Data

### DISCUSSION THREAD

**Discussion by Spieledmilk26** [sec-edgar/sec-edgar#283] [Q&A]
I want to pull 10-K reports and extract/save data to use for analysis in python. I was able to download a 10 K report, but I couldn't figure out how to save and utilise it. Pls help lol

**Comment by gyund** [comment]
I was searching around, and I came across [py-xbrl](https://py-xbrl.readthedocs.io/en/latest/usage.html#offline). As a test, I downloaded aapl's 10-K, and it had parsing errors. Mismatched XML and the like. I started looking to see if anything existed that covers #282 but nothing yet. I thought about using that data set instead to get at the numbers quicker and throw into something like pandas.

**Comment by jackmoody11** [comment]
> I want to pull 10-K reports and extract/save data to use for analysis in python. I was able to download a 10 K report, but I couldn't figure out how to save and utilise it. Pls help lol

The newest version of the package uses the REST API which returns actual numbers if you are just looking to pull financial data.
